community/kernel/src/main/java/org/neo4j/kernel/configuration/ServerConfigurationValidator.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.neo4j.kernel.configuration;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+
+import org.neo4j.graphdb.config.InvalidSettingException;
+import org.neo4j.graphdb.config.SettingValidator;
+import org.neo4j.logging.Log;
+
+import static org.neo4j.kernel.configuration.Connector.ConnectorType.BOLT;
+import static org.neo4j.kernel.configuration.Connector.ConnectorType.HTTP;
+
+
+public class ServerConfigurationValidator implements ConfigurationValidator
```

Comment:
Could that method be generalized into something that would allow the actual server-specific stuff to move out of Kernel? I'm worried that this and the move of the HTTP stuff down to sit alongside the Bolt stuff is making things worse in a way that we don't have a plan to improve.